### ISSUE TITLE
Shoot boat on other player claim

### ISSUE THREAD

**Issue by andris155** [TechFortress/GriefPrevention#850]
### What happened:
_What you saw_
If I shoot an arrow on a boat that is the other player claim flood the plugin no permission.

### What was expected:
_What you expected to see_
If I shoot an arrow on a boat that is the other player claim write once no permission.

### Steps to reproduce:
__This may include a build schematic, a video, or detailed instructions to help reconstruct the issue__
1. Shoot an arrow on a boat that is the other player claim
2. Flood no permission.

### Server and GriefPrevention version:
___This can be found by running `/version` and `/version GriefPrevention` on your server.___
Paste of `/version`: 
This server is running Paper version git-Paper-341 (MC: 1.15.2) (Implementing API version 1.15.2-R0.1-SNAPSHOT)

Paste of `/version GriefPrevention`: 
GriefPrevention version 16.13.0

### Stack trace/error or server log

```
[12:59:10] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:10] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:10] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:10] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:10] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:10] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:10] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:10] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:10] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:10] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:10] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:10] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:10] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:10] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:10] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:11] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:12] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:12] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:12] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:12] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:12] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:12] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:12] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:12] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:12] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:12] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:12] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:12] [main/INFO]: [CHAT] That belongs to andris155.
[12:59:12] [main/INFO]: [CHAT] That belongs to andris155.
```
Or, paste the server.log at gist.github.com and paste the link here: 

### GriefPrevention config.yml

```yml
# Default values are perfect for most servers.  If you want to customize and have a question, look for the answer here first: http://dev.bukkit.org/bukkit-plugins/grief-prevention/pages/setup-and-configuration/
GriefPrevention:
  SeaLevelOverrides:
    world2: -1
    world2_nether: -1
    world2_the_end: -1
    spawn2: -1
    plotworld: -1
    survivalgames: -1
    world114: -1
  Claims:
    Mode:
      world2_the_end: Survival
      spawn2: Disabled
      world2: Survival
      survivalgames: Disabled
      plotworld: Disabled
      world2_nether: Survival
      world114: Survival
    PreventGlobalMonsterEggs: true
    PreventTheft: true
    ProtectCreatures: true
    PreventButtonsSwitches: true
    LockWoodenDoors: false
    LockTrapDoors: false
    LockFenceGates: true
    EnderPearlsRequireAccessTrust: true
    ProtectHorses: true
    ProtectDonkeys: true
    ProtectLlamas: true
    InitialBlocks: 100
    Claim Blocks Accrued Per Hour:
      Default: 150
    Max Accrued Claim Blocks:
      Default: 200000
    Accrued Idle Threshold: 0
    AccruedIdlePercent: 0
    AbandonReturnRatio: 1.0
    AutomaticNewPlayerClaimsRadius: 4
    ExtendIntoGroundDistance: 5
    MinimumWidth: 5
    MinimumArea: 100
    MaximumDepth: 0
    InvestigationTool: GOLDEN_SWORD
    ModificationTool: GOLDEN_SHOVEL
    Expiration:
      ChestClaimDays: 14
      UnusedClaimDays: 28
      AllClaims:
        DaysInactive: 180
        ExceptWhenOwnerHasTotalClaimBlocks: 10000
        ExceptWhenOwnerHasBonusClaimBlocks: 5000
      AutomaticNatureRestoration:
        SurvivalWorlds: false
    AllowTrappedInAdminClaims: false
    MaximumNumberOfClaimsPerPlayer: 0
    CreationRequiresWorldGuardBuildPermission: true
    VillagerTradingRequiresPermission: false
    CommandsRequiringAccessTrust: /sethome
    DeliverManuals: true
    ManualDeliveryDelaySeconds: 30
    RavagersBreakBlocks: true
    FireSpreadsInClaims: false
    FireDamagesInClaims: false
    LecternReadingRequiresAccessTrust: true
  Spam:
    Enabled: false
    LoginCooldownSeconds: 60
    LoginLogoutNotificationsPerMinute: 5
    ChatSlashCommands: /me;/global;/local
    WhisperSlashCommands: /tell;/pm;/r;/whisper;/msg
    WarningMessage: Please reduce your noise level.  Spammers will be banned.
    BanOffenders: true
    BanMessage: Banned for spam.
    AllowedIpAddresses: 1.2.3.4; 5.6.7.8
    DeathMessageCooldownSeconds: 120
    Logout Message Delay In Seconds: 0
  PvP:
    RulesEnabledInWorld:
      world2: true
      world2_nether: true
      world2_the_end: true
      spawn2: true
      plotworld: false
      survivalgames: true
      world114: true
    ProtectFreshSpawns: false
    PunishLogout: false
    CombatTimeoutSeconds: 0
    AllowCombatItemDrop: true
    BlockedSlashCommands: /home;/vanish;/spawn;/tpa
    ProtectPlayersInLandClaims:
      PlayerOwnedClaims: true
      AdministrativeClaims: true
      AdministrativeSubdivisions: true
    AllowLavaDumpingNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: true
    AllowFlintAndSteelNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: true
    ProtectPetsOutsideLandClaims: false
  Economy:
    ClaimBlocksPurchaseCost: 150.0
    ClaimBlocksSellValue: 0.0
  ProtectItemsDroppedOnDeath:
    PvPWorlds: false
    NonPvPWorlds: false
  BlockLandClaimExplosions: false
  BlockSurfaceCreeperExplosions: false
  BlockSurfaceOtherExplosions: false
  LimitSkyTrees: true
  LimitTreeGrowth: false
  CheckPistonMovement: false
  LimitPistonsToLandClaims: false
  FireSpreads: true
  FireDestroys: true
  AdminsGetWhispers: false
  AdminsGetSignNotifications: false
  SmartBan: true
  Mute New Players Using Banned Words: false
  MaxPlayersPerIpAddress: 3
  Siege:
    Worlds: []
    BreakableBlocks:
    - DIRT
    - GRASS_BLOCK
    - GRASS
    - FERN
    - DEAD_BUSH
    - COBBLESTONE
    - GRAVEL
    - SAND
    - GLASS
    - GLASS_PANE
    - OAK_PLANKS
    - SPRUCE_PLANKS
    - BIRCH_PLANKS
    - JUNGLE_PLANKS
    - ACACIA_PLANKS
    - DARK_OAK_PLANKS
    - WHITE_WOOL
    - ORANGE_WOOL
    - MAGENTA_WOOL
    - LIGHT_BLUE_WOOL
    - YELLOW_WOOL
    - LIME_WOOL
    - PINK_WOOL
    - GRAY_WOOL
    - LIGHT_GRAY_WOOL
    - CYAN_WOOL
    - PURPLE_WOOL
    - BLUE_WOOL
    - BROWN_WOOL
    - GREEN_WOOL
    - RED_WOOL
    - BLACK_WOOL
    - SNOW
    DoorsOpenDelayInSeconds: 300
    CooldownEndInMinutes: 60
  EndermenMoveBlocks: true
  SilverfishBreakBlocks: true
  CreaturesTrampleCrops: true
  RabbitsEatCrops: true
  HardModeZombiesBreakDoors: true
  Database:
    URL: ''
    UserName: ''
    Password: ''
  UseBanCommand: false
  BanCommandPattern: ban %name% %reason%
  Advanced:
    fixNegativeClaimblockAmounts: true
    ClaimExpirationCheckRate: 60
    OfflinePlayer_cache_days: 90
  Abridged Logs:
    Days To Keep: 7
    Included Entry Types:
      Social Activity: false
      Suspicious Activity: false
      Administrative Activity: false
      Debug: false
      Muted Chat Messages: false
```

### Plugin list (if applicable):
GriefPrevention

### Debugging steps attempted

- [x] I attempted running only GriefPrevention on the server
  - [x] Issue still occurs even if GriefPrevention is the only plugin running
- [x] I attempted testing for the issue on a new server
  - [x] Issue still occurs even after testing on a new server

**Comment by Jikoo** [comment]
Thought of this issue while working on reorganizing some of damage handling. Turns out this problem is specific to vehicles or tridents.

When handling vehicles, GP does not deal with the projectile causing the warning at all.
When handling tridents, GP does not remove them because they're a pain to get. All other projectiles are removed.

I'll try to solve this as part of reorganizing damage handling - there's no reason why the EntityDamageByEntityEvent handling couldn't also handle the vehicle damage logic, which would also allow us to remove a lot of similar code.

In testing with entities and tridents, I've teleported the trident to the entity's feet and removed its velocity. It doesn't look quite as pretty as some fancy team-based collision prevention, but it works and it's way easier to implement.
